lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequestBuilder.cs

```diff
@@ -17,43 +17,98 @@
  * under the License.
  */
 
-using System.Collections.Generic;
-using Org.Apache.REEF.Common.Catalog;
-using Org.Apache.REEF.Common.Catalog.Capabilities;
+using System;
 
 namespace Org.Apache.REEF.Driver.Evaluator
 {
-    public class EvaluatorRequestBuilder
+    public sealed class EvaluatorRequestBuilder
     {
-        public EvaluatorRequestBuilder(EvaluatorRequest request)
+        private string _evaluatorBatchId;
+        private string _rackName;
+
+        [Obsolete("This constructor will be internal after 0.13.")]
+        public EvaluatorRequestBuilder(IEvaluatorRequest request)
         {
-            foreach (ICapability capability in request.Capabilities)
-            {
-                Capabilities.Add(capability);
-            }
             Number = request.Number;
-            Catalog = request.Catalog;
             MegaBytes = request.MemoryMegaBytes;
             VirtualCore = request.VirtualCore;
+            _evaluatorBatchId = request.EvaluatorBatchId;
+            _rackName = request.Rack;
         }
 
         internal EvaluatorRequestBuilder()
         {
+            Number = 1;
+            VirtualCore = 1;
+            MegaBytes = 64;
+            _rackName = String.Empty;
+            _evaluatorBatchId = Guid.NewGuid().ToString("N");
         }
 
         public int Number { get; set; }
```

Comment:
I'll remove them. I initially wanted to keep them for compatibility's sake. However, we decided to make these breaking changes :-)